pkg/executor/sortexec/sort.go

```diff
@@ -96,6 +97,13 @@ type SortExec struct {
 	enableTmpStorageOnOOM bool
 }
 
+// When fetcher and workers are not created, we need to initiatively close these channels
+func (e *SortExec) closeChannels() {
+	close(e.Parallel.resultChannel)
+	close(e.Parallel.chunkChannel)
+	close(e.Parallel.closeSync)
```

Comment:
why add this line here, does it means there is another bug?

Reply:
> why add this line here, does it means there is another bug?

We will wait for the `closeSync` only when `fetched` flag is set, so it's the same bug with the current issue.